scripts/test-update-chart-version.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+# Path to the script to be tested
+SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]:-"$0"}")")"
+SCRIPT_TO_TEST="$SCRIPT_DIR/update-chart-version.sh"
+
+# Function to run a test case
+run_test() {
+  local test_name=$1
+  local expected_output=$2
+  shift 2
+  local output
+
+  echo "Running: $test_name"
+  output=$("$SCRIPT_TO_TEST" "$@" 2>&1)
+  if [ "$output" == "$expected_output" ]; then
+    echo "PASS"
+  else
+    echo "FAIL"
+    echo "Expected: $expected_output"
+    echo "Got: $output"
+  fi
+  echo "----------------------------------------"
+}
+
+# Define test cases
+run_test "Test 1: Generate version from release branch name" \
```

Comment:
Does the test cater for when we are not on develop branch?

Reply:
Yes, if you see the caller of this test, it happens on all jobs.

Reply:
What I meant is, we shouldn't test with current Chart in workspace, we should add overrides and test for all cases, ex: develop and release, if you see what I mean?

Reply:
Okay. I added the test to run when we checkout develop as well. Would that be enough? Or you think we should add in the other workflows as well?